VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using VideoApi.DAL.Commands;
+using VideoApi.DAL.Commands.Core;
+using VideoApi.DAL.Queries;
+using VideoApi.DAL.Queries.Core;
+using VideoApi.Domain.Enums;
+using VideoApi.Events.Handlers.Core;
+using VideoApi.Events.Hub;
+using VideoApi.Events.Models;
+using VideoApi.Events.ServiceBus;
+using TaskStatus = VideoApi.Domain.Enums.TaskStatus;
+
+namespace VideoApi.Events.Handlers
+{
+    public class SelfTestFailedEventHandler : EventHandlerBase
+    {
+        public SelfTestFailedEventHandler(IQueryHandler queryHandler, ICommandHandler commandHandler,
+            IServiceBusQueueClient serviceBusQueueClient, IHubContext<EventHub, IEventHubClient> hubContext) : base(
+            queryHandler, commandHandler, serviceBusQueueClient, hubContext)
+        {
+        }
+
+        public override EventType EventType => EventType.SelfTestFailed;
+
+        protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
+        {
+            var query = new GetTasksForConferenceQuery(SourceConference.Id);
+            var tasks = await QueryHandler.Handle<GetTasksForConferenceQuery, List<Domain.Task>>(query);
+            var task = tasks.SingleOrDefault(x => x.Type == TaskType.Participant 
+                && x.OriginId == SourceParticipant.Id && x.Body == callbackEvent.Reason);
```

Comment:
include tasks that are ToDo too


`tasks.SingleOrDefault(x => x.Type == TaskType.Participant && x.Status == TaskStatus.ToDo
                && x.OriginId == SourceParticipant.Id && x.Body == callbackEvent.Reason);`

Reply:
updated to include the todo tasks as well.

Reply:
you seem to be missing a scenario to ensure the AddTaskCommand is not called when an existing pending task is found.

Reply:
added the test case to not call addTaskCommand if a task exists